### PR TITLE
Make Camera Settings

### PR DESCRIPTION
**Pull request by Ubaydulloh99**
Camera Settings component - done

image

### CODE REVIEW

File: qgroundcontrol.qrc [hunk 1]
```diff
@@ -183,7 +183,7 @@
         <file alias="QGroundControl/Controls/TerrainStatus.qml">src/PlanView/TerrainStatus.qml</file>
         <file alias="QGroundControl/Controls/TakeoffItemMapVisual.qml">src/PlanView/TakeoffItemMapVisual.qml</file>
         <file alias="QGroundControl/Controls/ToolStrip.qml">src/QmlControls/ToolStrip.qml</file>
-       	<file alias="QGroundControl/Controls/ToolStripHoverButton.qml">src/QmlControls/ToolStripHoverButton.qml</file>
+        <file alias="QGroundControl/Controls/ToolStripHoverButton.qml">src/QmlControls/ToolStripHoverButton.qml</file>
```
Comment (thanhkaist) [inline qgroundcontrol.qrc:186]: shorten the alias

File: src/QmlControls/CameraSettings.qml [hunk 2]
```diff
@@ -0,0 +1,337 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.14
+import QtPositioning 5.14
+
+Rectangle {
+    id: _root
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: remove _ do that id is root. This follows the naming convention

File: src/QmlControls/CameraSettings.qml [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+    color: '#000000'   // black
+
+    property var rootWidth: null;
+    property var rootHeight: null;
+
+    width: rootWidth / 3.24
+    height: rootHeight / 1
+
+    Text {
+        id: text1
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: rename the id to be move descriptive

File: src/QmlControls/CameraSettings.qml [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+        color: "#FFFFFF"    // white
+        text: qsTr("Camera Settings")
+        font.pixelSize: 25
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Using relative ScreenTools.defaultFontPixelHeight

File: src/QmlControls/CameraSettings.qml [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+                anchors.leftMargin: 15
+                text: 'Grid'
+                font.pixelSize: 20
+                color: "#FFFFFF"
+                // verticalAlignment: Text.AlignVCenter
+            }
+        }
+
+        ToolSeparator {
+            id: toolSeparator1
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: No need id for this

File: src/QmlControls/CameraSettings.qml [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+            height: 40
+            enabled: true
+            hoverEnabled: true
+
+            Text {
+                anchors.left: parent.left
+                anchors.leftMargin: 15
+                color: "#ffffff"
+                text: qsTr("Center Point")
+                font.pixelSize: 20
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Using ScreenTools.defaultFontPixelHeight

**Review by thanhkaist — CHANGES_REQUESTED** [review]
Checkout the https://github.com/argosdyne/AELS_QGC/blob/master/CodingStyle.qml and update the naming convention

File: qgroundcontrol.qrc (hunk 1 above)
Comment (Ubaydulloh99) [reply qgroundcontrol.qrc:186]: this ToolStripHoverButton row was already here, i cannot remove this line, only small open space was automatically entered i tried but it is not going away, it is staying there .i cannot delete this alias.

File: src/QmlControls/CameraSettings.qml [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+import QtQuick.Controls 2.14
+import QtPositioning 5.14
+
+Rectangle {
+    id: root
+    color: '#000000'   // black
+
+    property var rootWidth: null;
+    property var rootHeight: null;
+
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: The rootWidth and rootHeight properties are declared but not bound to any values. Consider binding these properties to the actual screen dimensions or relevant parent dimensions.

property int rootWidth: Screen.width
property int rootHeight: Screen.height

File: src/QmlControls/CameraSettings.qml [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+        font.family: 'Arial'
+        anchors.top: parent.top
+        anchors.left: parent.left
+        anchors.leftMargin: 10
+        anchors.topMargin: 12  // Add margin from top of the parent if needed
+    }
+
+    Column {
+        id: column1
+        anchors.top: text1.bottom
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: The anchors.top for the Column is set to text1.bottom, but text1 is undefined

File: src/QmlControls/CameraSettings.qml [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+        }
+
+        ToolSeparator {
+            width: parent.width
+            height: 13
+            orientation: Qt.Horizontal
+        }
+
+        ItemDelegate {
+            id: itemDelegate1
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: The IDs for ItemDelegate and SwitchDelegate components are unique but redundant if not referenced elsewhere. Remove unused IDs to reduce clutter.

File: src/QmlControls/CameraSettings.qml [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+        }
+
+        ToolSeparator {
+            width: parent.width
+            height: 13
+            orientation: Qt.Horizontal
+        }
+
+        SwitchDelegate {
+            id: switchDelegate
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: The IDs for ItemDelegate and SwitchDelegate components are unique but redundant if not referenced elsewhere. Remove unused IDs to reduce clutter.

File: src/QmlControls/CameraSettings.qml [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+        }
+
+        ToolSeparator {
+            width: parent.width
+            height: 13
+            orientation: Qt.Horizontal
+        }
+
+        SwitchDelegate {
+            id: switchDelegate0
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: The IDs for ItemDelegate and SwitchDelegate components are unique but redundant if not referenced elsewhere. Remove unused IDs to reduce clutter.

File: src/QmlControls/CameraSettings.qml [hunk 12]
```diff
@@ -0,0 +1,326 @@
...
+        }
+
+        ToolSeparator {
+            width: parent.width
+            height: 13
+            orientation: Qt.Horizontal
+        }
+
+        SwitchDelegate {
+            id: switchDelegate1
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: The IDs for ItemDelegate and SwitchDelegate components are unique but redundant if not referenced elsewhere. Remove unused IDs to reduce clutter.

File: src/QmlControls/CameraSettings.qml [hunk 13]
```diff
@@ -0,0 +1,326 @@
...
+        }
+
+        ToolSeparator {
+            width: parent.width
+            height: 13
+            orientation: Qt.Horizontal
+        }
+
+        SwitchDelegate {
+            id: switchDelegate2
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: The IDs for ItemDelegate and SwitchDelegate components are unique but redundant if not referenced elsewhere. Remove unused IDs to reduce clutter.

File: src/QmlControls/CameraSettings.qml [hunk 14]
```diff
@@ -0,0 +1,326 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.14
+import QtPositioning 5.14
+
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml:6]: You may miss to include screenTools

import QGroundControl.ScreenTools   1.0

File: src/QmlControls/CameraSettings.qml [hunk 15]
```diff
@@ -0,0 +1,326 @@
...
+        SwitchDelegate {
+            id: switchDelegate
+            width: parent.width
+            height: 52
+            text: qsTr(" ")
+
+            Text {
+                anchors.left: parent.left
+                anchors.leftMargin: 15
+                color: "#ffffff"
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Text color are all #ffffff. Could you consider using a centralized theme or constants for colors to maintain consistency and ease of updates.

**Review by thanhkaist — COMMENTED** [review]
Could you consider to improve these suggestion

File: src/QmlControls/CameraSettings.qml [hunk 16]
```diff
@@ -0,0 +1,306 @@
...
+        anchors.topMargin: 15 // Editable margin between text1 and column1
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.bottom: parent.bottom
+
+        spacing: 4
+
+        ItemDelegate {
+            width: parent.width
+            height: 40
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 17]
```diff
@@ -0,0 +1,306 @@
...
+                anchors.leftMargin: 15
+                text: 'Grid'
+                font.pixelSize: ScreenTools.mediumFontPointSize * 2
+                color: white
+            }
+        }
+
+        ToolSeparator {
+            width: parent.width
+            height: 13
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 18]
```diff
@@ -0,0 +1,306 @@
...
+            orientation: Qt.Horizontal
+        }
+
+        ItemDelegate {
+            width: parent.width
+            height: 40
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 19]
```diff
@@ -0,0 +1,306 @@
...
+                anchors.leftMargin: 15
+                color: white
+                text: qsTr("Center Point")
+                font.pixelSize: ScreenTools.mediumFontPointSize * 2
+            }
+        }
+
+        ToolSeparator {
+            width: parent.width
+            height: 13
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 20]
```diff
@@ -0,0 +1,306 @@
...
+            orientation: Qt.Horizontal
+        }
+
+        SwitchDelegate {
+            width: parent.width
+            height: 52
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 21]
```diff
@@ -0,0 +1,306 @@
...
+                anchors.leftMargin: 15
+                color: white
+                text: qsTr("Histogram")
+                font.pixelSize: ScreenTools.mediumFontPointSize * 2
+            }
+        }
+
+        ToolSeparator {
+            width: parent.width
+            height: 13
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 22]
```diff
@@ -0,0 +1,306 @@
...
+            orientation: Qt.Horizontal
+        }
+
+        SwitchDelegate {
+            width: parent.width
+            height: 52
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 23]
```diff
@@ -0,0 +1,306 @@
...
+                anchors.leftMargin: 15
+                color: white
+                text: qsTr("Lock Gimbal While Shooting")
+                font.pixelSize: ScreenTools.mediumFontPointSize * 2
+            }
+        }
+
+        ToolSeparator {
+            width: parent.width
+            height: 13
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 24]
```diff
@@ -0,0 +1,306 @@
...
+            orientation: Qt.Horizontal
+        }
+
+        SwitchDelegate {
+            width: parent.width
+            height: 52
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 25]
```diff
@@ -0,0 +1,306 @@
...
+                anchors.leftMargin: 15
+                color: white
+                text: qsTr("DeFog")
+                font.pixelSize: ScreenTools.mediumFontPointSize * 2
+            }
+        }
+
+        ToolSeparator {
+            width: parent.width
+            height: 13
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 26]
```diff
@@ -0,0 +1,306 @@
...
+            orientation: Qt.Horizontal
+        }
+
+        SwitchDelegate {
+            width: parent.width
+            height: 52
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

File: src/QmlControls/CameraSettings.qml [hunk 27]
```diff
@@ -0,0 +1,306 @@
...
```
Comment (thanhkaist) [inline src/QmlControls/CameraSettings.qml]: Hardcoding. Please use property binding

**Review by thanhkaist — COMMENTED** [review]
Use property binding for the height

**Review by thanhkaist — APPROVED** [review]

**Pull request merged**